pkg/reconciler/metrics/metrics.go

```diff
@@ -0,0 +1,240 @@
+
+	triggersclientset "github.com/tektoncd/triggers/pkg/client/clientset/versioned"
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/pkg/logging"
+	"knative.dev/pkg/metrics"
+)
+
+var (
+	elMetricsName = "eventlistener_count"
+	elCount       = stats.Float64(elMetricsName,
+		"number of eventlistener",
+		stats.UnitDimensionless)
+	elCountView *view.View
+
+	tbMetricsName = "triggerbinding_count"
+	tbCount       = stats.Float64(tbMetricsName,
+		"number of triggerbinding",
+		stats.UnitDimensionless)
+	tbCountView *view.View
+
+	ctbMetricsName = "clustertriggerbinding_count"
+	ctbCount       = stats.Float64(ctbMetricsName,
+		"number of clustertriggerbinding",
+		stats.UnitDimensionless)
+	ctbCountView *view.View
+
+	ttMetricsName = "triggertemplate_count"
+	ttCount       = stats.Float64(ttMetricsName,
+		"number of triggertemplate",
+		stats.UnitDimensionless)
+	ttCountView *view.View
+
+	ciMetricsName = "clusterinterceptor_count"
+	ciCount       = stats.Float64(ciMetricsName,
+		"number of clusterinterceptor",
+		stats.UnitDimensionless)
+	ciCountView *view.View
+)
+
+// Recorder holds information for Trigger metrics
+type Recorder struct {
+	initialized     bool
+	client          triggersclientset.Interface
+	ReportingPeriod time.Duration
+}
+
+// We cannot register the view multiple times, so NewRecorder lazily
+// initializes this singleton and returns the same recorder across any
+// subsequent invocations.
+var (
+	once        sync.Once
+	r           *Recorder
+	recorderErr error
+)
+
+// NewRecorder creates a new metrics recorder instance
+// to log the PipelineRun related metrics
+func NewRecorder(ctx context.Context) (*Recorder, error) {
+	once.Do(func() {
+		r = &Recorder{
+			initialized: true,
+			client:      triggersclient.Get(ctx),
+			// Default to reporting metrics every 60s.
+			ReportingPeriod: 60 * time.Second,
+		}
+
+		recorderErr = viewRegister()
+		if recorderErr != nil {
+			r.initialized = false
+			return
+		}
+	})
+
+	return r, recorderErr
+}
+
+func viewRegister() error {
+	elCountView = &view.View{
+		Description: elCount.Description(),
+		Measure:     elCount,
+		Aggregation: view.LastValue(),
+	}
+
+	tbCountView = &view.View{
+		Description: tbCount.Description(),
+		Measure:     tbCount,
+		Aggregation: view.LastValue(),
+	}
+
+	ctbCountView = &view.View{
+		Description: ctbCount.Description(),
+		Measure:     ctbCount,
+		Aggregation: view.LastValue(),
+	}
+
+	ttCountView = &view.View{
+		Description: ttCount.Description(),
+		Measure:     ttCount,
+		Aggregation: view.LastValue(),
+	}
+
+	ciCountView = &view.View{
+		Description: ciCount.Description(),
+		Measure:     ciCount,
+		Aggregation: view.LastValue(),
+	}
+
+	return view.Register(
+		elCountView,
+		tbCountView,
+		ctbCountView,
+		ttCountView,
+		ciCountView,
+	)
+}
+
+func (r *Recorder) CountMetrics(ctx context.Context) {
```

Comment:
Could we write a test for this function?

Reply:
I added a separate function which uses listers, I can testing it.